controllers/openstackipset_controller.go

```diff
@@ -485,37 +449,47 @@ func (r *OpenStackIPSetReconciler) addIP(instance *ospdirectorv1beta1.OpenStackI
 					IPnet:           *cidr,
 					RangeStart:      start,
 					RangeEnd:        end,
-					RoleReservelist: network.Status.RoleReservations[instance.Spec.RoleName].Reservations,
+					RoleReservelist: network.Spec.RoleReservations[instance.Spec.RoleName].Reservations,
 					Reservelist:     reservationList,
 					ExcludeRanges:   []string{},
 					Hostname:        hostname,
 					VIP:             instance.Spec.VIP,
 					Deleted:         false,
 				})
 				if err != nil {
-					return fmt.Errorf(fmt.Sprintf("Failed to do ip reservation: %v", err))
+					cond.Message = fmt.Sprintf("Failed to do ip reservation: %s", hostname)
+					cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.IPSetCondReasonIPReservation)
+					cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.IPSetCondTypeError)
+					err = common.WrapErrorForObject(cond.Message, instance, err)
+
+					return ctrl.Result{}, err
 				}
 
 				reservationIP = ip.String()
 
 				// record the reservation on the OpenStackNet
-				network.Status.RoleReservations[instance.Spec.RoleName] = ospdirectorv1beta1.OpenStackNetRoleStatus{
+				network.Spec.RoleReservations[instance.Spec.RoleName] = ospdirectorv1beta1.OpenStackNetRoleStatus{
 					AddToPredictableIPs: instance.Spec.AddToPredictableIPs,
 					Reservations:        reservation,
 				}
 
-				r.Log.Info(fmt.Sprintf("Created new reservation for host %s in network %s with IP %s", hostname, netName, reservationIP))
-
+				common.LogForObject(
+					r,
+					fmt.Sprintf("Created new reservation for host %s in network %s with IP %s", hostname, netName, reservationIP),
+					instance,
+				)
 			}
 
 			// add netName -> reservationIP to hostIPs map
 			hostNetworkIPs[netName] = reservationIP
 
-			err = r.Client.Status().Update(context.TODO(), network)
+			// update status of OSNet
+			// TODO: just patch?
```

Comment:
I know you put this `TODO` in a few places, but I just had a thought on it.  It seems that either would be fine here, but it's easier, I think, to just do the `Update` if you already have the whole object.  Maybe `Patch` can dodge the `"objected has been modified"` error in some cases though?

Reply:
yes, I agree that for now we should be good to just update the full object. I was more thinking on when used server side apply there is a field management mechanism and if a different manager updates a field it not owns it will result in a conflict. if we get to this and see it, we might need to use patch to just update a single field.

Reply:
Yeah, I think it would be interesting to investigate using `Patch` here (but not right now) and elsewhere in an effort to reduce the occurrences of conflicts